iothub/device/src/Transport/AmqpIot/AmqpIotResourceException.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport.AmqpIot
 {
-    internal class AmqpIotResourceException : IotHubException
+    internal class AmqpIotResourceException : IotHubClientException
```

Comment:
Do you plan on deprecating this exception type?

Reply:
No. This one is for internally usage only, so I think deprecating it is sort of deviating from the purpose of this PR. I prefer to leave it as-is, but I am open to other thoughts.